### PR TITLE
Testoptions and websockets, now updated to be SSL compatible.

### PR DESCRIPTION
**Pull request by pboothe**
Pull request 2 of N.

### CODE REVIEW

File: src/websocket.c [hunk 1]
```diff
@@ -260,16 +261,16 @@ int64_t recv_websocket_ndt_msg(int socket_fd, int* msg_type, char* msg_value,
  * Reads a line from the filedescriptor, up to maxlen long. Returns the length
```
Comment (mtlynch) [inline src/websocket.c]: `filedescriptor` needs update

File: src/websocket.c [hunk 2]
```diff
@@ -355,40 +361,42 @@ int send_digest_base64(int socket_fd, const unsigned char* message_digest) {
...
   // Neither of these limits is in the HTTP spec, but they are the limits
   // enforced by the Apache defaults. In particular, it is recommended that
   // there not be more than 8K of headers total, so these limits are generous.
   // http://stackoverflow.com/questions/686217/maximum-on-http-header-values
-  const static unsigned int MAX_HEADER_COUNT = 1024;
-  const static unsigned int MAX_HEADER_LENGTH = 8192;
+  // They are #define'd because declaring arrays of variable size, even if the
+  // variable is const and static, is a gcc extension and not part of C.
+  #define MAX_HEADER_COUNT 1024
+  #define MAX_HEADER_LENGTH 8192
```
Comment (mtlynch) [inline src/websocket.c]: Optional:
* `MAX_HEADER_COUNT` can still be an int
* To me it's less gross to just move 383 into line 373's spot as `line[8192]` and just do `sizeof` to get the value back when needed. I think it's actually better to do it that way even if you don't get rid of `MAX_HEADER_LENGTH`. e.g.:

```
ws_readline(conn, line, sizeof(line))
```

File: src/websocket.c [hunk 3]
```diff
@@ -469,17 +482,17 @@ int write_websocket_header(int socket_fd, char* protocol, char* key) {
  *                          May be NULL or "" if no protocol name is desired.
  * @returns 0 on success, error code otherwise.
  */
-int initialize_websocket_connection(int socket_fd, unsigned int skip_bytes,
+int initialize_websocket_connection(Connection* conn, unsigned int skip_bytes,
```
Comment (mtlynch) [inline src/websocket.c]: Need to update function comment

**Comment by mtlynch** [comment]
LGTM

File: src/websocket.c (hunk 1 above)
Comment (pboothe) [inline src/websocket.c]: Done

File: src/websocket.c (hunk 3 above)
Comment (pboothe) [inline src/websocket.c]: done

File: src/websocket.c (hunk 2 above)
Comment (pboothe) [inline src/websocket.c]: Used 8192 and sizeof

**Comment by pboothe** [comment]
All cleaned up.

**Pull request merged**